src/policy_evaluator.rs

```diff
@@ -16,12 +16,76 @@ use crate::cluster_context::ClusterContext;
 use crate::policy::Policy;
 use crate::validation_response::ValidationResponse;
 
+#[derive(Clone, PartialEq)]
 pub enum PolicyExecutionMode {
     KubewardenWapc,
     Opa,
     OpaGatekeeper,
 }
 
+impl serde::ser::Serialize for PolicyExecutionMode {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::ser::Serializer,
+    {
+        let mode = match self {
+            PolicyExecutionMode::KubewardenWapc => "kubewarden",
+            PolicyExecutionMode::Opa => "opa",
+            PolicyExecutionMode::OpaGatekeeper => "gatekeeper",
+        };
+        serializer.serialize_str(mode)
+    }
+}
+
+struct StringVisitor;
+
+impl<'de> serde::de::Visitor<'de> for StringVisitor {
+    type Value = String;
+
+    fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+        write!(formatter, "a string")
+    }
+
+    fn visit_str<E>(self, s: &str) -> Result<Self::Value, E>
+    where
+        E: serde::de::Error,
+    {
+        Ok(s.to_owned())
+    }
+}
```

Comment:
Do we need this pattern?

Reply:
You need it if you want to implement a custom Derive. However, it turns out we don't need it, more below